lang_scala/parsing/Parser_scala_recursive_descent.ml

```diff
@@ -3565,27 +3565,28 @@ let compilationUnit in_ : top_stat list =
   (* ast:  case ... makeEmptyPackage ... *)
   topstats in_
 
-let parse toks =
+
+let try_rule toks frule =
   let in_ = mk_env toks in
   init in_;
-  let xs = compilationUnit in_ in
+  let x = frule in_ in
   accept (EOF ab) in_;
-  xs
+  x
+
+let parse toks =
+  try
+    try_rule toks compilationUnit
+  with PI.Parsing_error _ ->
+    try_rule toks block
```

Comment:
suggestion: maybe a comment saying officially scala accepts only full compilationUnit but in practice
lots of scripts contain just a block.